### PR TITLE
Issue #12601: adds tests for whitespace mutations

### PR DESCRIPTION
**Pull request by rnveach**
Issue #12601

### CODE REVIEW

**Comment by rnveach** [comment]
No regression needed.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheckTest.java [hunk 1]
```diff
@@ -164,6 +173,29 @@ public void testGenericWhitespaceWithEmoji() throws Exception {
                 getPath("InputGenericWhitespaceWithEmoji.java"), expected);
     }
 
+    @Test
+    public void testClearState() throws Exception {
```
Comment (nrmancuso) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheckTest.java:177]: Please add a comment here explaining why we need synthetic test.

**Review by nrmancuso — CHANGES_REQUESTED** [review]

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheckTest.java (hunk 1 above)
Comment (rnveach) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheckTest.java:177]: Done.

Comment (romani) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheckTest.java:183]: I am ok, we can be fanatic on elimination of such test in next phase to let us focus on pitest now.

**Review by romani — APPROVED** [review]
ok to merge

**Review by nrmancuso — APPROVED** [review]

**Pull request merged**